File: CuoiKy2012_Bai4_DiemLoiNhoNhat/main.py
import sys
sys.stdin = open("1.inp", "r")
sys.stdout = open("1.out", "w")

class SubNode:
    def __init__(self):
        self.value = float('inf')
        self.i = float('inf')
        self.j = float('inf')
        self.is_valid = False

def create_matrix(n, m, a):
    for i in range(1, n + 1):
        row = list(map(float, input().split()))
        for j in range(1, m + 1):
            a[i][j] = row[j - 1]

def handle(n, m, a, s):
    s.value = float('inf')
    s.i = float('inf')
    s.j = float('inf')
    for i in range(1, n + 1):
        for j in range(1, m + 1):
            top = a[i - 1][j]
            bottom = a[i + 1][j]
            left = a[i][j - 1]
            right = a[i][j + 1]
            if a[i][j] > top and a[i][j] > bottom and a[i][j] > left and a[i][j] > right:
                if a[i][j] < s.value:
                    s.value = a[i][j]
                    s.i = i - 1
                    s.j = j - 1
                    s.is_valid = True

def show_sub_node(s):
    if s.is_valid == True:
        print(int(s.i), int(s.j))
    else:
        print(-1)

def main():
    n, m = map(int, input().split())
    v = [[float('-inf')] * (m + 2) for _ in range(n + 2)]
    s = SubNode()
    create_matrix(n, m, v)
    handle(n, m, v, s)
    show_sub_node(s)

if __name__ == "__main__":
    main()

File: CuoiKy2012_Bai4_DiemLoiNhoNhat/main.cpp
#include <bits/stdc++.h>
using namespace std;
#define READ(name)                                                             \
  freopen(name ".inp", "r", stdin);                                            \
  freopen(name ".out", "w", stdout);
#define ll long double
struct sub_node {
  ll value;
  ll i, j;
  bool is_valid = false;
};

void create_matrix(ll n, ll m, vector<vector<ll>> &a);
void show_matrix(ll n, ll m, vector<vector<ll>> &a);
void handle(ll n, ll m, vector<vector<ll>> &a, sub_node &s);
void show_sub_node(sub_node &s);
int main() {
  ios_base::sync_with_stdio(false);
  cin.tie(NULL);
  cout.tie(NULL);
  READ("1");
  ll n, m;
  cin >> n >> m;
  vector<vector<ll>> v(n + 2, vector<ll>(m + 2, (ll)-1e18));
  sub_node s;
  create_matrix(n, m, v);
  handle(n, m, v, s);
  show_sub_node(s);
  return 0;
}

void create_matrix(ll n, ll m, vector<vector<ll>> &a) {
  for (int i = 1; i <= n; i++) {
    for (int j = 1; j <= m; j++) {
      cin >> a[i][j];
    }
  }
}
void handle(ll n, ll m, vector<vector<ll>> &a, sub_node &s) {
  s.value = (ll)1e18;
  s.i = (ll)1e18;
  s.j = (ll)1e18;
  for (int i = 1; i <= n; i++) {
    for (int j = 1; j <= m; j++) {
      ll top = a[i - 1][j];
      ll bottom = a[i + 1][j];
      ll left = a[i][j - 1];
      ll right = a[i][j + 1];
      if (a[i][j] > top && a[i][j] > bottom && a[i][j] > left &&
          a[i][j] > right) {
        if (a[i][j] < s.value) {
          s.value = a[i][j];
          s.i = i - 1;
          s.j = j - 1;
          s.is_valid = true;
        }
      }
    }
  }
}

void show_sub_node(sub_node &s) {
  if (s.is_valid == true) {
    cout << s.i << " " << s.j << endl;
  } else {
    cout << "-1" << endl;
  }
}
